music21/midi/translate.py

```diff
@@ -1739,9 +1740,14 @@ def getMetaEvents(events):
         elif e.type == MetaEvents.SET_TEMPO:
             metaObj = midiEventsToTempo(e)
         elif e.type in (MetaEvents.INSTRUMENT_NAME, MetaEvents.SEQUENCE_TRACK_NAME):
+            # midiEventsToInstrument() WILL NOT have knowledge of the current
+            # program, so set it here
             metaObj = midiEventsToInstrument(e)
+            metaObj.midiProgram = last_program
```

Comment:
`if last_program != -1`  -- so we're trusting midiEventsToInstrument if no program was set.